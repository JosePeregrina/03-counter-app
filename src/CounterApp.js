import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
  
  const [ counter, setCounter ] = useState(value);

  // handleAdd
  const handleAdd = (e) => {
    // Add one to the counter
    setCounter( (counter)=> counter + 1 )
  };

  const handleReset = () => {
    //Return the counter to its otiginal value
    setCounter(value)
  }

  const hanldeSubstract = () => {
    //Substract one to the counter
    setCounter((counter) => counter - 1 )
  }

  return (
    <>
      <h1> CounterApp </h1>
      <h2> {counter} </h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={hanldeSubstract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
