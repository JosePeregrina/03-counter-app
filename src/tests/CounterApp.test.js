import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('El componente CounterApp', () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test('cargarse correctamente con sus valores por defecto "hacer snapshot"', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de mostar el valor por defecto de 100', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const counterValue = Number.parseInt(wrapper.find('h2').text());

    expect(counterValue).toBe(value);
  });

  test('debe de incrementar con el botón de +1 ', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterValue = Number.parseInt(wrapper.find('h2').text());

    expect(counterValue).toBe(11);
  });

  test('debe de decrementar con el botón de -1 ', () => {
    wrapper.find('button').at(2).simulate('click');
    const counterValue = Number.parseInt(wrapper.find('h2').text());

    expect(counterValue).toBe(9);
  });

  test('debe de resetear el valor con el botón de reset ', () => {
    const wrapper = shallow(<CounterApp value={15} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const counterValue = Number.parseInt(wrapper.find('h2').text());

    expect(counterValue).toBe(15);
  })
  
});
