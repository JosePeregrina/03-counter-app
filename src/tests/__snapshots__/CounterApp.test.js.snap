// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente CounterApp cargarse correctamente con sus valores por defecto "hacer snapshot" 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
